sympy/core/numbers.py

```diff
@@ -1745,93 +1733,93 @@ def __eq__(self, other):
             other = _sympify(other)
         except SympifyError:
             return NotImplemented
+        if other.is_zero:
+            return self.is_zero
+        if self.is_zero:
+            return other.is_zero
         if other.is_NumberSymbol:
             if other.is_irrational:
                 return False
             return other.__eq__(self)
-        if other.is_Number:
-            if other.is_Rational:
-                # a Rational is always in reduced form so will never be 2/4
-                # so we can just check equivalence of args
-                return self.p == other.p and self.q == other.q
-            if other.is_Float:
-                return mlib.mpf_eq(self._as_mpf_val(other._prec), other._mpf_)
+        if other.is_Rational:
+            # a Rational is always in reduced form so will never be 2/4
+            # so we can just check equivalence of args
+            return self.p == other.p and self.q == other.q
+        if other.is_Float:
```

Comment:
@jksuom , this is where all the Float/Rational comparison logic takes place. I think I experimented with comparing them in an approximate way and returning True if they differed by less than the range permitted by the precision.

Reply:
I would start with a quick test to see if the denominator is a power of 2:

    if self.q & (self.q - 1):
        return False

(Then we also know that `integer_log(self.q, 2)` will return True.)

Reply:
beautiful! done.